docs/go/testing.md

```diff
@@ -183,3 +183,59 @@ function matches the signature of the original Activity function.
 
 Since this can be an entire function, there is no limitation as to what we can do here. In this
 example, we assert that the `value` param has the same content as the value param we passed to the Workflow.
+
+## Queries
+
+`TestWorkflowEnvironment` instances have a [`QueryWorkflow()` method](https://pkg.go.dev/go.temporal.io/temporal/internal#TestWorkflowEnvironment.QueryWorkflow) that lets you query the currently running Workflow.
+For example, suppose you have a Workflow that lets you query the progress of a long running task as shown below.
+
+```go
+func ProgressWorkflow(ctx workflow.Context, percent int) error {
+	logger := workflow.GetLogger(ctx)
+
+	err := workflow.SetQueryHandler(ctx, "getProgress", func(input []byte) (int, error) {
+		return percent, nil
+	})
+	if err != nil {
+		logger.Info("SetQueryHandler failed.", "Error", err)
+		return err
+	}
+
+	for percent = 0; percent<100; percent++ {
+		workflow.Sleep(ctx, time.Second*1)
+	}
+
+	return nil
+}
+```
+
+This Workflow tracks the current progress of a task in percentage terms, and increments the percentage by 1 every second.
+Below is how you would write a test case that queries this Workflow.
+Note that you should always query the Workflow either after `ExecuteWorkflow()` is done or in a `RegisterDelayedCallback()` callback, otherwise you'll get a `runtime error` panic.
+
+```go
+func (s *UnitTestSuite) Test_ProgressWorkflow() {
+	value := 0
+
+	// After 10 seconds plus some padding, the progress should be 10.
+	// Note that `RegisterDelayedCallback()` doesn't actually make your test wait for 10 seconds!
+	// This test should take far less than 1 second.
+	s.env.RegisterDelayedCallback(func() {
+		res, err := s.env.QueryWorkflow("getProgress")
+		s.NoError(err)
+		err = res.Get(&value)
+		s.NoError(err)
+		s.Equal(10, value)
+	}, time.Second*10+time.Millisecond*2)
+
+	s.env.ExecuteWorkflow(ProgressWorkflow, 0)
+
+	s.True(s.env.IsWorkflowCompleted())
+
+	res, err := s.env.QueryWorkflow("getProgress")
+	s.NoError(err)
+	err = res.Get(&value)
+	s.NoError(err)
+	s.Equal(value, 100)
```

Comment:
you should probably add comment here that you are querying the workflow after its in the Completed status so its clear to readers